translate.go

```diff
@@ -0,0 +1,135 @@
+package i18n
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/getlantern/golog"
+	"github.com/getlantern/jibber_jabber"
+)
+
+// ReadFunc is the func provided to WillReadByFunc
+// which returns the byte sequence given a file name
+type ReadFunc func(fileName string) ([]byte, error)
+
+var (
+	localeRegexp  string   = "^[a-z]{2}([_-][A-Z]{2}){0,1}$"
+	log                    = golog.LoggerFor("i18n")
+	readFunc      ReadFunc = makeReadFunc("locale")
+	defaultLocale string   = "en_US"
```

Comment:
Oh sorry I forgot to mention that.
I just can't think of a scene when user sets locale to en_US or something else, but falls back to zh_CN. Maybe we can add it later.
How do you think?